common.blocks/demo/demo.js

```diff
@@ -65,37 +64,43 @@ modules.define('demo', [ 'i-bem__dom', 'pretty', 'functions__debounce' ], functi
             }
             this._result.setValue(finalCode);
         },
-        _save: function() {
+        save: function() {
             var template = this._getTemplate(),
-                bemjson = this._getBEMJSON();
+                bemjson = this._getBEMJSON(),
+                version = this._versionSelect.getValue(),
+                engine = this._engineSelect.getValue();
 
             store.set('playground', {
-                version: this.params.version,
+                version: version,
                 template: template,
-                bemjson: bemjson
+                bemjson: bemjson,
+                engine: engine
             });
 
             history.pushState({}, document.title, location.pathname + '?' + [
                 'template=' + encodeURIComponent(template),
-                'bemjson=' + encodeURIComponent(bemjson)
+                'bemjson=' + encodeURIComponent(bemjson),
+                'version=' + encodeURIComponent(version),
+                'engine=' + encodeURIComponent(engine)
             ].join('&'));
         },
         _load: function() {
-            var data = parseParams(location.search.split('?')[1]) || store.get('playground');
-
-            if (!data || (data.version && data.version !== this.params.version)) {
-                return false;
-            }
-
-            if (typeof data.template === 'string') {
-                this._templates.setValue(data.template);
-            }
-
-            if (typeof data.bemjson === 'string') {
-                this._bemjson.setValue(data.bemjson);
-            }
-
-            return !!(data.template || data.bemjson)
+            var data = parseParams(location.search.split('?')[1]) ||
+                store.get('playground') ||
+                this._getDefaultState();
+
+            this._templates.setValue(data.template);
+            this._bemjson.setValue(data.bemjson);
+            this._versionSelect.setValue(data.version);
+            this._engineSelect.setValue(data.engine);
```

Comment:
Smart decision, sir! :) Just didn't want to perform it in sake of diff sanity, but now it surely has to be done.